### ISSUE TITLE
Multi MySQL Server Support

### ISSUE THREAD

**Issue by ImJacobChen** [leokhoa/laragon#240]
Hi there,

First of all thank you for building laragon. It has been extremely helpful for me over the past years.

Now, is it possible that support can be added to run multiple MySQL servers at the same time on different ports? I need this functionality for local development. Currently I run additional MySQL servers via windows command line alongside the laragon MySQL server but would love to have all servers in laragon.

If there is any way I can help bring this feature live please let me know.

Many thanks.

Jacob

**Comment by bcs02123052** [comment]
Yeah it would be great if @leokhoa  you can add multi MySQL support sometimes i have to work with MySQL 5.7 and sometimes i have to work with MySQL 8 I have to delete the MySQL data directory before switching

**Comment by ImJacobChen** [comment]
> Yeah it would be great if @leokhoa you can add multi MySQL support sometimes i have to work with MySQL 5.7 and sometimes i have to work with MySQL 8 I have to delete the MySQL data directory before switching

Hey @bcs02123052, I think for your use case you would have the two directories such as
 - `C:\laragon\bin\mysql\mysql-5.7.24-winx64`
 - `C:\laragon\bin\mysql\mysql-8.0.21-winx64`

and then you can use `Menu -> Mysql -> Version -> *The version you require*`.. Unless you want both versions running at the same time?

Which is kind of what I require. I require three separate instances running on separate ports at the same time

**Comment by bcs02123052** [comment]
> > Yeah it would be great if @leokhoa you can add multi MySQL support sometimes i have to work with MySQL 5.7 and sometimes i have to work with MySQL 8 I have to delete the MySQL data directory before switching
> 
> Hey @bcs02123052, I think for your use case you would have the two directories such as
> 
> * `C:\laragon\bin\mysql\mysql-5.7.24-winx64`
> * `C:\laragon\bin\mysql\mysql-8.0.21-winx64`
> 
> And also edit the `my.ini` for each installation to point to `laragon/data/mysql57` & `laragon/data/mysql8` respectively.
> 
> and then you can use `Menu -> Mysql -> Version -> *The version you require*`... Unless you want both versions running at the same time?
> 
> Which is kind of what I require. I require three separate instances running on separate ports at the same time

Thanks you for this awesome trick 👍

**Comment by JaservTECH** [comment]
how about running mysql and maria db in same time with different port can laragon do this or maybe have another trick because i need 2 database server same time

**Comment by borhanuddin** [comment]
@JaservTECH 
MariaDB and MySQL have almost identical structure, you can put all your db in either MariaDB or MySQL

**Comment by JaservTECH** [comment]
Thanks bro, i have research yes its identical, option is maria or mysql , noted thanks bro ()
________________________________

**Comment by JaservTECH** [comment]
Sorry for contacting you again bro but i have server using laragon but i need to know how passing parameter on laragon .ini
I've upgrade my redis version using cygwin got problem on running cause laragon cannot found redis.conf
I've research that need passing custom parameter
still has error when launch from laragon · Issue #22 · redis-windows/redis-windows (github.com)<https://github.com/redis-windows/redis-windows/issues/22>
Unable to start redis built on cygwin from laragon · Issue #753 · leokhoa/laragon (github.com)<https://github.com/leokhoa/laragon/issues/753>

from

E:\Server\bin\redis\redis-x64-7.2.5\redis-server.exe E:\Server\bin\redis\redis-x64-7.2.5\redis.conf

to

E:\Server\bin\redis\redis-x64-7.2.5\redis-server.exe /cygdrive/E\Server\bin\redis\redis-x64-7.2.5\redis.conf

But on laragon.ini only got
Version=.....
QuickSetting=.....
Use=....

Is there any way to add the command
I know it can run by terminal or cmd directly but i hope i can easily just one click on start all service if it possible
thanks bro
________________________________

**Comment by leokhoa** [comment]
You can have multiple instances of Laragon which runs different MySQL server on different ports.
